examples/configmap_reflector.rs

```diff
@@ -12,7 +12,11 @@ fn spawn_periodic_reader(reader: Store<ConfigMap>) {
         loop {
             // Periodically read our state
             tokio::time::delay_for(std::time::Duration::from_secs(5)).await;
-            let cms: Vec<_> = reader.state().iter().map(|obj| Meta::name(obj).clone()).collect();
+            let cms: Vec<_> = reader
+                .state()
+                .iter()
+                .map(|obj| Meta::name(obj).to_string())
```

Comment:
Think this should be able to borrow it? So just `.map(Meta::name)`.

Reply:
No, that doesn't work unfortunately. "Temporary value dropped while burrowed".

Reply:
Oh, aw. I guess something is off with our lifetime bounds somewhere? :/

Reply:
I think this is just the standard issue where we are trying to collect references to strings that are only available in the scope up. `reader` is dropped at the end of `collect`, and then you just have a vector of dangling slices.

Reply:
`reader` is still valid, but now that you mention it, it is correct that the handle expires with the `Iterator`.. :/

Reply:
Ah, yeah, sorry. the `state()` temporary value is dropped.

Reply:
Still nice enough to use by caching `reader.state()` done now.